cmd/occollector/app/sender/jaeger_proto_grpc_sender.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sender
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+
+	jaegerproto "github.com/jaegertracing/jaeger/proto-gen/api_v2"
+
+	"github.com/census-instrumentation/opencensus-service/data"
+	jaegertranslator "github.com/census-instrumentation/opencensus-service/translator/trace/jaeger"
+)
+
+// JaegerProtoGRPCSender forwards spans encoded in the jaeger proto
+// format, to a grpc server.
+type JaegerProtoGRPCSender struct {
+	client jaegerproto.CollectorServiceClient
+	logger *zap.Logger
+}
+
+// NewJaegerProtoGRPCSender returns a new GRPC-backend span sender.
+// The collector endpoint should be of the form "hostname:14250".
+func NewJaegerProtoGRPCSender(collectorEndpoint string, zlogger *zap.Logger) *JaegerProtoGRPCSender {
+	client, _ := grpc.Dial(collectorEndpoint, grpc.WithInsecure())
+	collectorServiceClient := jaegerproto.NewCollectorServiceClient(client)
+	s := &JaegerProtoGRPCSender{
+		client: collectorServiceClient,
+		logger: zlogger,
+	}
+
+	return s
+}
+
+// ProcessSpans sends the batch to the configured Jaeger Proto-GRPC endpoint.
+func (s *JaegerProtoGRPCSender) ProcessSpans(td data.TraceData, spanFormat string) error {
+	protoBatch, err := jaegertranslator.OCProtoToJaegerProto(td)
+	if err != nil {
+		return err
+	}
+
+	_, err = s.client.PostSpans(context.Background(), &jaegerproto.PostSpansRequest{Batch: *protoBatch})
```

Comment:
does `s` takes care of logging messages in case of errors?

Reply:
Ah, I don't seem to have used the logger, I'll fix this.

Reply:
Perhaps the jaeger code already uses it?

Reply:
This part of code doesn't seem to be passing the logger to `PostSpans()` (which just returns the error, we could log this with `s.logger`).

Reply:
ah, correct: when I looked I was thinking that s was from jaeger repo... ops...